src/ngx_http_lua_log_ringbuf.c

```diff
@@ -0,0 +1,206 @@
+
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+
+#include "ngx_http_lua_common.h"
+#include "ngx_http_lua_log_ringbuf.h"
+
+
+typedef struct {
+    unsigned    is_data:1;
+    unsigned    log_level:4;
+    int         len;
```

Comment:
Beware of padding, this layout is very inefficient. Please always put larger struct fields before smaller ones. Here you are doing the opposite. Alas.